src/harness/testcases/WINNF_FT_S_GPR_testcase.py

```diff
@@ -0,0 +1,455 @@
+      open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_request_1['operationParam']['operationFrequencyRange']['lowFrequency'] = 3650000000
+    grant_request_1['operationParam']['operationFrequencyRange']['highFrequency'] = 3655000000
+
+    grant_request_2 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_1.json')))
+    grant_request_2['operationParam']['operationFrequencyRange']['lowFrequency'] = 3660000000
+    grant_request_2['operationParam']['operationFrequencyRange']['highFrequency'] = 3665000000
+
+    # device_b is Category B
+    # Load Conditional Data
+    self.assertEqual(device_2['cbsdCategory'], 'B')
+    conditionals_device_2 = {
+        'cbsdCategory': device_2['cbsdCategory'],
+        'fccId': device_2['fccId'],
+        'cbsdSerialNumber': device_2['cbsdSerialNumber'],
+        'airInterface': device_2['airInterface'],
+        'installationParam': device_2['installationParam'],
+        'measCapability': device_2['measCapability']
+    }
+
+    conditionals = [conditionals_device_2]
+
+    # Remove conditionals from registration
+    del device_2['cbsdCategory']
+    del device_2['airInterface']
+    del device_2['installationParam']
+    del device_2['measCapability']
+
+    # Load GWPZ Record
+    gwpz_record_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+
+    devices = [device_1, device_2]
+    grants = [grant_request_1, grant_request_2]
+
+    # Create the actual config.
+    config = {
+        'registrationRequests': devices,
+        'grantRequests': grants,
+        'conditionalRegistrationData': conditionals,
+        'gwpzRecord': gwpz_record_1
+      }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_GPR_3_default_config)
+  def test_WINNF_FT_S_GPR_3(self, config_filename):
+    """GWPZ Exclusion Zone
+      One or more CBSDs, located inside a GWPZ, request Grants and are rejected
+    """
+    config = loadConfig(config_filename)
+
+    # Step1: Admin Test Harness injects a GWPZ into the SAS UUT
+    self._sas_admin.InjectWisp(config['gwpzRecord'])
+
+    # Step2: Admin Test Harness triggers CPAS and waits until completion of CPAS
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Step3: DP Test Harness registers N > 0 CBSDs
+    cbsd_ids = self.assertRegistered(config['registrationRequests'], config['conditionalRegistrationData'])
+
+    # Step4: DP Test Harness requests one Grant per registered CBSD
+    grant_request = config['grantRequests']
+    addCbsdIdsToRequests(cbsd_ids, grant_request)
+
+    request = {'grantRequest': grant_request}
+    response = self._sas.Grant(request)['grantResponse']
+    self.assertEqual(len(response), len(grant_request))
+
+    for response_num, response in enumerate(response):
+      self.assertEqual(response['cbsdId'], request['grantRequest'][response_num]['cbsdId'])
+      self.assertTrue('grantId' in response)
```

Comment:
You can't expect a grant ID if you're also expecting the request to fail.

Reply:
Yes you are right.
Removed the check for grant ID